environment/src/main/java/com/sequenceiq/environment/environment/flow/modify/proxy/config/EnvProxyModificationFlowConfig.java

```diff
@@ -0,0 +1,73 @@
+package com.sequenceiq.environment.environment.flow.modify.proxy.config;
+
+import static com.sequenceiq.environment.environment.flow.modify.proxy.EnvProxyModificationState.FINAL_STATE;
+import static com.sequenceiq.environment.environment.flow.modify.proxy.EnvProxyModificationState.INIT_STATE;
+import static com.sequenceiq.environment.environment.flow.modify.proxy.EnvProxyModificationState.PROXY_CONFIG_MODIFICATION_FAILED_STATE;
+import static com.sequenceiq.environment.environment.flow.modify.proxy.EnvProxyModificationState.PROXY_CONFIG_MODIFICATION_FINISHED_STATE;
+import static com.sequenceiq.environment.environment.flow.modify.proxy.EnvProxyModificationState.PROXY_CONFIG_MODIFICATION_START_STATE;
+import static com.sequenceiq.environment.environment.flow.modify.proxy.event.EnvProxyModificationStateSelectors.FAILED_MODIFY_PROXY_EVENT;
+import static com.sequenceiq.environment.environment.flow.modify.proxy.event.EnvProxyModificationStateSelectors.HANDLE_FAILED_MODIFY_PROXY_EVENT;
+
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.environment.environment.flow.modify.proxy.EnvProxyModificationState;
+import com.sequenceiq.environment.environment.flow.modify.proxy.event.EnvProxyModificationStateSelectors;
+import com.sequenceiq.flow.core.config.AbstractFlowConfiguration;
+import com.sequenceiq.flow.core.config.RetryableFlowConfiguration;
+
+@Component
+public class EnvProxyModificationFlowConfig extends
+        AbstractFlowConfiguration<EnvProxyModificationState, EnvProxyModificationStateSelectors>
+        implements RetryableFlowConfiguration<EnvProxyModificationStateSelectors> {
+
+    private static final List<Transition<EnvProxyModificationState, EnvProxyModificationStateSelectors>> TRANSITIONS =
+            new Transition.Builder<EnvProxyModificationState, EnvProxyModificationStateSelectors>()
+                    .defaultFailureEvent(FAILED_MODIFY_PROXY_EVENT)
+
+                    .from(INIT_STATE).to(PROXY_CONFIG_MODIFICATION_START_STATE)
+                    .event(EnvProxyModificationStateSelectors.MODIFY_PROXY_START_EVENT).defaultFailureEvent()
```

Comment:
Please be conform with other flow configs and do a fully qualified import for these enum values.